mmv1/products/vertexai/api.yaml

```diff
@@ -997,3 +998,74 @@ objects:
           The update method to use with this Index. The value must be the followings. If not set, BATCH_UPDATE will be used by default.
           * BATCH_UPDATE: user can call indexes.patch with files on Cloud Storage of datapoints to update.
           * STREAM_UPDATE: user can call indexes.upsertDatapoints/DeleteDatapoints to update the Index and the updates will be applied in corresponding DeployedIndexes in nearly real-time.
+
+# Vertex AI Matching Engine Index Endpoint
+  - !ruby/object:Api::Resource
+    name: IndexEndpoint
+    base_url: projects/{{project}}/locations/{{region}}/indexEndpoints
+    create_url: projects/{{project}}/locations/{{region}}/indexEndpoints
+    self_link: projects/{{project}}/locations/{{region}}/indexEndpoints/{{name}}
+    update_verb: :PATCH
+    update_mask: true
+    create_verb: :POST
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      api: https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.indexEndpoints/
+    async: !ruby/object:Api::OpAsync
+      operation: !ruby/object:Api::OpAsync::Operation
+        path: 'name'
+        base_url: '{{op_id}}'
+        wait_ms: 1000
+      result: !ruby/object:Api::OpAsync::Result
+        path: 'response'
+        resource_inside_response: true
+      status: !ruby/object:Api::OpAsync::Status
+        path: 'done'
+        complete: True
+        allowed:
+          - True
+          - False
+      error: !ruby/object:Api::OpAsync::Error
+        path: 'error'
+        message: 'message'
+    description: |-
+      An endpoint indexes are deployed into. An index endpoint can have multiple deployed indexes.
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: region
+        description: The region of the index endpoint. eg us-central1
+        url_param_only: true
+        input: true
+    properties:
```

Comment:
Any reason for not including `deployedIndexes[]`? https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.indexEndpoints#DeployedIndex

Reply:
Thank you for your review! A reason is the `deployedIndexes[]` is the output-only field. So I didn't define it here. 

Another reason is I plan to create `google_vertex_ai_deployed_index` to manage a deployed index. I'll use the `deployIndexes[]` when implementing it for the read operation. The description of https://github.com/hashicorp/terraform-provider-google/issues/12818 has the ideas behind `google_vertex_ai_deployed_index` and its corresponding API endpoints.

What do you think?

Reply:
Yep, this looks good to me, thanks!